### PR TITLE
docs

### PR DESCRIPTION
**Pull request by ozgunozerk**
Adds documentation

Note: `cargo doc` is currently broken. The problem is fixed, and hopefully, tomorrow we will have a new nightly release.
I don't want to merge this since I'll update the toolchain accordingly after testing `cargo doc` works fine.

However, this is not a blocker for the review process :)

### CODE REVIEW

**Comment by ozgunozerk** [comment]
We are good to go if we are to use `cargo doc --no-deps` one. 
If we are not to supply `--no-deps` flag, the issue is still there and we should wait for another release for the nightly @achiurizo

**Review by i1i1 — APPROVED** [review]

**Pull request merged**
